subprojects/core-api/src/main/java/org/gradle/api/artifacts/Configuration.java

```diff
@@ -382,6 +383,18 @@ public String determineName(Configuration c) {
      */
     DependencyConstraintSet getAllDependencyConstraints();
 
+    /**
+     * Use the given collector as a source for dependencies and dependency constraints.
+     *
+     * @param collector the collector to use
+     * @since 8.7
+     */
+    @Incubating
+    default void useDependencyCollector(DependencyCollector collector) {
```

Comment:
At the very least I would prefer `fromDependencyCollector` over `useDependencyCollector`, though I like `from` the most I think since it seems very "gradley" -- though it is also a very non-descriptive name. `dependenciesFrom` is also not so bad IMO. 

I would avoid overloading `withDependencies` since that method already exists on `Configuration` for another purpose. `addAllLater` overloads the method on NDOC which could be confusing.

Reply:
+1 to `fromDependencyCollector`

Reply:
Also +1 to `addAllLater`

Reply:
Replaced with `fromDependencyCollector`.